src/librustc_trans/debuginfo/mod.rs

```diff
@@ -168,7 +168,7 @@ pub fn finalize(cx: &CrateContext) {
     }
 
     debug!("finalize");
-    let _ = compile_unit_metadata(cx);
+    //let _ = compile_unit_metadata(cx);
```

Comment:
It's indeed possible that the updated code (that creates the metadata early) will work with older LLVMs. Specifically, LLVM inverted the way the debug information graph is built. All the builders inside LLVM magically handle this, **as long as there is something to add the debug information to**. Creating the debug information as early as possible enables this.